node/src/syncer.rs

```diff
@@ -366,6 +366,7 @@ where
 }
 
 #[cfg(test)]
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
In which place Syncer tests panic under WASM. I think they shouldn't

Reply:
It's the sleep in `MockP2pHandle::expect_cmd`, they panic when accessing time: 
```
        panicked at library/std/src/sys/wasm/../unsupported/time.rs:13:9:
        time not implemented on this platform
```

Reply:
I see. This is because of `timeout` I used in expect_cmd. You need to create the equivalent function for wasm the same signature and have it under `executor.rs`

Reply:
That fixed the `time` panic, additionally I've changed command receiving part of syncer: https://github.com/eigerco/celestia-node-rs/pull/102/files#diff-4d0e726dde3f3ba66c0cd97c8f61c7aa1fc2104ca66e3d04d2bda1e488909930R174-R179

Worker thread wasn't being killed at the end of tests and causing them to never finish. I've changed it so that if command channel is dropped, worked exists.